### PR TITLE
chore: Month component

### PR DESCRIPTION
**Pull request by gpbl**
In the context of #1311, extract a new component rendering the month from the `Root` component.

### CODE REVIEW

File: packages/react-day-picker/src/components/Month/Month.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+
+import { useDayPicker, useNavigation } from '../../contexts';
+import { Table } from '../Table';
+
+/** The props for the [[Month]] component. */
+export interface MonthProps {
+  displayIndex: number;
+  displayMonth: Date;
```
Comment (tjfred35) [inline packages/react-day-picker/src/components/Month/Month.tsx:9]: Seems a bit weird to pass in both of these and then also query for `displayMonths`...perhaps we only pass in displayMonth or displayIndex, and we compute the other based on displayMonths?

File: packages/react-day-picker/src/components/Month/Month.test.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import { customRender } from '@test/render';
+
+import { Month, MonthProps } from './Month';
+import { DayPickerProps } from '../../types';
+
+let root: HTMLDivElement;
+function setup(props: MonthProps, dayPickerProps: DayPickerProps) {
+  const renderResult = customRender(<Month {...props} />, dayPickerProps);
+  root = renderResult.container.firstChild as HTMLDivElement;
+  console.log(root.style);
```
Comment (tjfred35) [inline packages/react-day-picker/src/components/Month/Month.test.tsx:12]: Do we want this here?

**Review by tjfred35 — APPROVED** [review]
LGTM

File: packages/react-day-picker/src/components/Month/Month.test.tsx (hunk 2 above)
Comment (gpbl) [reply packages/react-day-picker/src/components/Month/Month.test.tsx:12]: Sure no, added the eslint rule. Thanks!

File: packages/react-day-picker/src/components/Month/Month.tsx (hunk 1 above)
Comment (gpbl) [reply packages/react-day-picker/src/components/Month/Month.tsx:10]: I'd prefer we keep `displayMonth` and remove `displayIndex`, then get the index from the hoo. 

Still I have no idea how to write the test: I think I should mock the contex. I'd keep it like it is for this PR: being internal component shouldn't affect too much. Feel free to give a look to see if we can remove one prop.

**Pull request merged**